HAR-6897 Hyväksy hinnoittelu

Kauko opened the pull request.

Kauko committed:
Lisää toimenpiteen hinnoittelu tietokantaan

Tauluun tallennetaan rivi joka kerta, kun hinnoittelua
muokataan tai hyväksytään/hylätään.

Kauko committed:
Hae ja tallenna toimenpiteen kommentteja

Kauko committed:
Päivitä gridin sarakkeiden leveyksiä, lisää tila-sarake

Kauko committed:
Lisää napit hinnoittelun hyväksynnälle, hylkäämiselle

Kauko committed:
Järjestä riippuvuudet

Kauko committed:
Poista turha laskutus_test.cljs

Kauko committed:
Ota maksueriin, laskutusyhteenvedolle vain hyväksytyt hinnoittelut

Viewiin otetaan toimenpide-id:t, joiden uusimman kommentin tila on
'hyväksytty'.

Kauko committed:
Lisää ilmaa hinnoittelun hyväksyntänappeihin

Kauko committed:
Muuta nappien ulkoasuja

Kauko committed:
Merge branch 'develop' into HAR-6897

Kauko committed:
Poista poistettu laskutus_test myös runnerista

Kauko committed:
Lisää testidataan hyväksytty ja käsittelemätön hinnoittelu, testien takia

Jarzka committed:
Merge branch 'develop' into HAR-6897

Kauko committed:
Merge branch 'develop' into HAR-6897

Kauko committed:
Merge branch 'develop' into HAR-6897

Jarzka commented on tietokanta/src/main/resources/db/migration/V1_670__.sql:
```diff
@@ -0,0 +1,25 @@
+CREATE TYPE kommentin_tila AS ENUM ('luotu', 'muokattu', 'poistettu', 'hyvaksytty', 'hylatty');
+
+CREATE TABLE kan_toimenpide_kommentti (
```
Mietin aluksi, että miksi tässä ei käytetä kommentti-taulua, mutta vaikuttaa aika spesiaalilta tapaukselta, ehkä ehkä oma tietomalli on ok.

Jarzka commented on src/cljs/harja/views/kanavat/urakka/toimenpiteet/hinnoittelu.cljs:
```diff
@@ -298,34 +416,22 @@
-                                                                    (:id @nav/valittu-urakka))))}
-         :arvo (fmt/euro-opt (nykyisten-arvo app* toimenpide-rivi valittu-aikavali))
-         :ikoninappi? true}))]))
+       [:div.arvo-ja-nappi-container
```
Miksi ei enää käytetä arvo-ja-nappi komponenttia? Johtuuko siitä, että tässä tehdään arvo ja useampi nappi? Ehkä se siinä tapauksessa on ok.

Jarzka commented on src/cljs/harja/views/kanavat/urakka/toimenpiteet/hinnoittelu.cljs:
```diff
@@ -280,11 +278,131 @@
+
+(defn hyvaksy-tai-hylkaa [e! app* hinta toimenpide-rivi]
+  [:div.arvo-ja-nappi
+   [:span.arvo-ja-nappi-arvo
```
Miksi ei käytetä arvo-ja-nappi komponenttia? Selkeästi näyttäisi siltä, että tässä tehdään käsin arvo ja nappi?

Jarzka commented on src/cljs/harja/views/kanavat/urakka/toimenpiteet/hinnoittelu.cljs:
```diff
@@ -280,11 +278,131 @@
+     :luokka (str "btn-xs arvo-ja-nappi-nappi")
+     :disabled (or (:id (:infolaatikko-nakyvissa app*))
+                   (not (oikeudet/on-muu-oikeus? "hinnoittele-toimenpide"
+                                                 oikeudet/urakat-vesivaylatoimenpiteet-yksikkohintaiset
```
Sama kysymys kuin ylempänä, miksi ei arvo-ja-nappi komponenttia?

Jarzka committed:
Levennä rivinvalintasaraketta

Kauko committed:
Merge branch 'develop' into HAR-6897

Kauko committed:
Käytä grid/arvo-ja-nappi

Samalla lisätty jvh-roolille hieman erikoiskäsittelyä,
jotta näkymää voi testailla jotenkuten vain jvh:na.

Koko workflow vaatii kuitenkin kahden roolin välillä
vaihtamisen.

Jarzka committed:
Rename V1_672__.sql to V1_673__.sql
